native/main.ts

```diff
@@ -163,6 +164,26 @@ ipcMain.handle(RendererMessage.OPEN_URL, (_event, url) => {
   openExternalLink(url);
 });
 
+ipcMain.handle(RendererMessage.GET_GIT_GLOBAL_DEFAULT_BRANCH, async () => {
+  return await new Promise((resolve, reject) =>
+    childProcess.exec(
+      "git config --global --get init.defaultBranch",
+      {
+        encoding: "utf-8",
+      },
+      (error, stdout, stderr) => {
+        if (error) {
+          reject(error);
```

Comment:
This throws now.Either we wrap the whole promise in `try/catch` or we don’t reject here.

Reply:
So my idea initially was to try to get the git config. If it failed, the only expectable reason would be the running git version not supporting it yet. So we'd either get it or the config is undefined, thus the `string | undefined` return type I had initially. In other terms, `Maybe<string>`.

When we moved to use the async alternative, since it returns a `Promise`, I felt divided between `Promise<string | undefined>` (aka `Promise<Maybe<string>>`) or capturing the error lane as a `reject` and see it more like an async `Either`.

Since we are not interested in the particular error, using a `Maybe`-like approach seems appropriate in the end. Meaning that we could wrap the whole promise in a `try/catch` and return `undefined` when it throws.

wdyt?

Reply:
> Meaning that we could wrap the whole promise in a `try/catch` and return `undefined` when it throws.

Sounds good

Reply:
Done!